tests/utils/test_tensors.py

```diff
@@ -211,3 +211,23 @@ def test_remove():
         pt.remove(idx, 1)
     # removes one page per batch-dim
     assert free_pages + pt.shape[0] == pt.paged_storage.free_pages()
+
+
+def test_compile():
```

Comment:
You may want to add something here similar to https://github.com/foundation-model-stack/foundation-model-stack/blob/c4d154cd0dec3d336b67a59cecdee1a23247785e/fms/testing/_internal/model_test_suite.py#L164 (testing for graph breaks)

Reply:
hmmm ... is that just allowing enforcement of single-graph through the `assert cnt.frame_count...`? Won't `full_graph=True` fail if there are graph breaks anyway?